### PR TITLE
fix: use _malloc/_free to allocate memory in setMemory

### PR DESCRIPTION
**Pull request by mnater**
see issue #5595

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,12 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Release")
 endif()
 
+# Optionally change stack size (e.g.: emcmake cmake . -Dbinaryen_STACK_SIZE=8388608)
+if(NOT binaryen_STACK_SIZE)
```
Comment (kripken) [inline CMakeLists.txt:31]: Looks like the other options are in the form `BYN_` so please make this `BYN_STACK_SIZE`. Or, as this is only for binaryen.js/wasm, perhaps `BYN_WASM_STACK_SIZE`.

**Review by kripken — COMMENTED** [review]
malloc/free looks good!

Please remove the other changes, though - I don't think we need to make the stack size customizable now. (And it has a downside as it bakes 64K as the default, rather than deferring to emscripten to set the default.)

**Comment by mnater** [comment]
I don't know a clean way to remove the older commits. 🤷‍♂️
I'll close this PR and send a new one.

**Comment by mnater** [comment]
#5646

**Pull request closed without merging**